core/src/main/java/org/apache/iceberg/deletes/BitmapPositionDeleteIndex.java

```diff
@@ -92,4 +107,113 @@ public Collection<DeleteFile> deleteFiles() {
   public long cardinality() {
     return bitmap.cardinality();
   }
+
+  /**
+   * Serializes the index using the following format:
+   *
+   * <ul>
+   *   <li>The length of the magic bytes and bitmap stored as 4 bytes (big-endian).
+   *   <li>A 4-byte {@link #MAGIC_NUMBER} (little-endian).
+   *   <li>The bitmap serialized using the portable Roaring spec (little-endian).
+   *   <li>A CRC-32 checksum of the magic bytes and bitmap as 4-bytes (big-endian).
+   * </ul>
+   *
+   * Note that the length and the checksum are computed for the bitmap data, which includes the
+   * magic bytes and bitmap for compatibility with Delta.
+   */
+  @Override
+  public ByteBuffer serialize() {
+    bitmap.runLengthEncode(); // run-length encode the bitmap before serializing
```

Comment:
Should we more eagerly run length encode the bitmap in the constructor instead of at serialization time?

Reply:
NVM, I don't think it'll make any difference and probably more readable the way it's written at the moment.

Reply:
We actually want to accumulate all values prior to applying run-length encoding, so doing this right before the serialization seems like a good place. We don't anticipate the bitmap to change after serialization.